tensorflow_probability/python/optimizer/bfgs.py

```diff
@@ -197,10 +197,10 @@ def quadratic(x):
                                        name='x_tolerance')
     max_iterations = tf.convert_to_tensor(max_iterations, name='max_iterations')
 
-    domain_shape = initial_position.shape
+    domain_shape = tf.shape(initial_position)
 
     if initial_inverse_hessian_estimate is None:
-      inv_hessian_shape = domain_shape.concatenate(domain_shape)
+      inv_hessian_shape = tf.concat((domain_shape, domain_shape), 0)
```

Comment:
Please use a list instead of a tuple for consistency with the rest of the code base.